src/CommonLib/DirectoryObjects/DirectoryObjectExtensions.cs

```diff
@@ -35,15 +35,15 @@ public static class DirectoryObjectExtensions {
             return false;
         }
         
-        if (!directoryObject.TryGetIntProperty(LDAPProperties.Flags, out var flags)) {
+        if (!directoryObject.TryGetLongProperty(LDAPProperties.Flags, out var flags)) {
             flags = 0;
         }
 
         directoryObject.TryGetDistinguishedName(out var distinguishedName);
         directoryObject.TryGetProperty(LDAPProperties.SAMAccountType, out var samAccountType);
         directoryObject.TryGetArrayProperty(LDAPProperties.ObjectClass, out var objectClasses);
 
-        return LdapUtils.ResolveLabel(objectIdentifier, distinguishedName, samAccountType, objectClasses, flags,
+        return LdapUtils.ResolveLabel(objectIdentifier, distinguishedName, samAccountType, objectClasses, (int)flags,
```

Comment:
Do we want to keep this as an int?

Reply:
Flags is always 0,1,or 2, so should be safe